scripts/02_reporting/extract-fiss-species-table.R

```diff
@@ -1,69 +1,74 @@
-                   INNER JOIN
-                   whse_basemapping.fwa_watershed_groups_poly ws
-                   ON ST_intersects(x.geom, ws.geom)
-                   WHERE ws.watershed_group_code IN
-                             ('MORR', 'ZYMO', 'KISP')")
-
-
-##lets bust it up and join it back together
-fish_spp <- merge(merge(fish_species_watershed %>% filter(watershed_group_code == 'MORR') %>% rename(Morice = watershed_group_code),
-                        fish_species_watershed %>% filter(watershed_group_code == 'ZYMO')%>% rename(Zymoetz = watershed_group_code),
-                        by = c('species_code', 'species_name'), all = TRUE),
-                        fish_species_watershed %>% filter(watershed_group_code == 'KISP')%>% rename(Kispiox = watershed_group_code),
-                        by = c('species_code', 'species_name'), all = TRUE
-                        )
-
-fish_all <- fishbc::freshwaterfish
-fish_cdc <- fishbc::cdc
-
-fish_spp2 <- left_join(fish_spp,
-                       fish_all,
-                       by = c("species_code" = "Code")) %>%
-  filter(!is.na(Class) & !species_code == 'TR') %>% ##mottled sculpin has some sort of error going on
-  # mutate(CDCode = case_when(species_code == 'BT' ~ 'F-SACO-11', ##pacific population yo
-  #                           T ~ CDCode)) %>%
-  select(species_code, species_name, Morice, Zymoetz, Kispiox, CDCode)
-
-fish_spp3 <- left_join(
-  fish_spp2,
-  fish_cdc,
-  by = c('CDCode' = 'Species Code')
-) %>%
-  select(`Scientific Name`,
-         'Species Name' = species_name,
-         'Species Code' = species_code,
-         `BC List`,
-         `Provincial FRPA`,
-         COSEWIC,
-         SARA,
-         Morice,
-         Zymoetz,
-         Kispiox) %>%
-  mutate(Morice = case_when(!is.na(Morice) ~ 'Yes',
-                                 T ~ Morice),
-         Zymoetz = case_when(!is.na(Zymoetz) ~ 'Yes',
-                              T ~ Zymoetz),
-         Kispiox = case_when(!is.na(Kispiox) ~ 'Yes',
-                          T ~ Kispiox)) %>%
-  arrange(`Scientific Name`, `Species Name`)
-
-##print your table to input_raw for use in the report
-fish_spp3 %>% readr::write_csv(file = paste0(getwd(), '/data/inputs_extracted/fiss_species_table.csv'))
-
-dbDisconnect(conn = conn)
+source('scripts/packages.R')
+
+##get the observations from the fiss layer
+fish_species_watershed <- fpr::fpr_db_query(query = "SELECT DISTINCT ws.watershed_group_code, x.species_code,x.species_name
+                   FROM whse_fish.fiss_fish_obsrvtn_pnt_sp x
+                   INNER JOIN
+                   whse_basemapping.fwa_watershed_groups_poly ws
+                   ON ST_intersects(x.geom, ws.geom)
+                   WHERE ws.watershed_group_code IN ('MORR', 'ZYMO', 'KISP', 'BULK', 'BULK')")
+
+
+#put all data frames into list
+fish_spp_prep <- list(fish_species_watershed %>% filter(watershed_group_code == 'MORR')%>% rename(Morice = watershed_group_code),
```

Comment:
best way to do this will be using a xref table.

Before was thinking we could have a function but then realized that could lead to unnecessary functions since building the xref tibbles is quite simple multiple ways.  See examples of how to build the cross reference table that can be generated in each repo here https://github.com/NewGraphEnvironment/rfp/issues/5

these calls to `merge` were done by Mateo and led to this issue https://github.com/NewGraphEnvironment/onboarding/issues/48 . Probably really annoying for you so sorry about that - but relevant none the less.